code/game/objects/items/devices/scanners.dm

```diff
@@ -65,14 +65,26 @@ GENE SCANNER
 	if(!ismob(viewer) || !viewer.client)
 		return
 	var/list/t_ray_images = list()
-	for(var/obj/O in orange(distance, viewer) )
-		if(HAS_TRAIT(O, TRAIT_T_RAY_VISIBLE))
-			var/image/I = new(loc = get_turf(O))
-			var/mutable_appearance/MA = new(O)
-			MA.alpha = 128
-			MA.dir = O.dir
-			I.appearance = MA
-			t_ray_images += I
+	for(var/turf/turf_to_scan as anything in RANGE_TURFS(distance, viewer))
+		var/list/list_to_scan = turf_to_scan.nullspaced_contents
```

Comment:
why cache this if we don't use it later?

Reply:
i think it was more to name it but its dumb